lib/api/apiUtils/object/objectRestore.js

```diff
@@ -142,6 +143,12 @@ function objectRestore(metadata, mdUtils, userInfo, request, log, callback) {
             if (isObjectRestored) {
                 monitoring.promMetrics(
                     'POST', bucketName, '200', 'restoreObject');
+                // If quotas are evaluated, increasing the quota duration should
+                // not affect the utilization metrics.
+                if (request.removeQuotaEvaluationInflights?.[mdValueParams]) {
+                    return request.removeQuotaEvaluationInflights?.[mdValueParams.requestType]?.(
+                        () => callback(null, 200, responseHeaders));
```

Comment:
instead of calling the cleanup callback, it would be cleaner to forcefully disable the quota check from `standardMetadataValidateBucketAndObj` (for this case only, e.g. add `disableQuota: true` at line 63), and thus explicitely call `validateQuota()` from here only when needed...

Reply:
The API was updated: quota check is disabled during authz, then we evaluate it manually just before updating the MD, and if the object state matches. THis also ensures we return `InvalidObjectState` before `QuotaExceeded`. I had to rework the tests, so that we edit the object MD using a metadata client (here, mongodb) to really test a restore.